### ISSUE TITLE
S3796: ignore returned Promise on async functions

### ISSUE THREAD

**Issue by fxOne** [SonarSource/SonarJS#895]
**I want to report a bug.**

**SonarJS version:** 3.3 (Build 5702)<br>
**SonarQube version:** 6.7 (Build 33306)<br>

**Rule key:** javascript:S3796

**Reproducer** 
```
  return Promise.all([1,2].map(async (int) => { // False-positive here
    await asyncFunctionCall(int);
  }));
```
**Expected behavior**
There shouldn't be an Tickets as async function would return a Promise even if there is no return a the function.
